examples/10_recipe.py

```diff
@@ -0,0 +1,451 @@
+# We are now done building our |Recipe|, and we can try it on some data and see
+# how it performs.
+#
+# As mentioned at the top of the page, a |Recipe| is not a scikit-learn
+# estimator with the usual ``fit`` and ``transform`` methods. Rather, it is a
+# way to configure such an object.
+# We must call one of the recipe's methods for getting the scikit-learn object
+# we have configured.
+#
+# We can use ``recipe.get_pipeline()`` which will return a scikit-learn
+# |Pipeline|. This object applies all the transformation steps we have added
+# and the final supervised learner. However, it performs no hyperparameter
+# selection. Wherever we specified a choice, it will just use the default value
+# (the first option for ``choose_from``, or the middle of the range for
+# ``choose_float`` and ``choose_int``).
+#
+# To perform hyperparameter selection, we can use ``recipe.get_grid_search()``
+# (returns a |GridSearchCV|) or ``recipe.get_randomized_search()`` (returns a
+# |RandomizedSearchCV|). Both run a nested loop of cross-validation on their
+# training data to select the best hyperparameters, then refit the best
+# combination of hyperparameters on the whole training data. The difference is
+# that the grid search will try all possible combinations, whereas the
+# randomized search will sample a few possible combinations (we decide how
+# many).
+#
+# See the scikit-learn `documentation
+# <https://scikit-learn.org/stable/modules/grid_search.html>`_ for more
+# information about hyperparameter search.
+#
+# Here we have specified a continuous range of values for the learning rate, so
+# "all possible values" does not make sense and we will sample a few instead:
+# we use the randomized search.
+
+# %%
+from sklearn.metrics import r2_score
+
+randomized_search = recipe.get_randomized_search(n_iter=16, cv=3, verbose=1)
+randomized_search.fit(recipe.get_x_train(), recipe.get_y_train())
+
+predictions = randomized_search.predict(recipe.get_x_test())
+score = r2_score(recipe.get_y_test(), predictions)
+print(f"R² score: {score:.2f}")
+
+# %%
+# We can use ``get_x_train()``, ``get_x_test()`` to get a default train/test
+# split of our data. If we wanted the whole data instead, for example to run a
+# cross-validation loop with ``sklearn.model_selection.cross_validate``, we
+# could use ``get_x()`` and ``get_y()``.
+#
+
+# %%
+# Inspecting hyperparameter search results
+# ----------------------------------------
+# Here we fitted a single model instead of running a cross-validation so that
+# we could inspect the results stored during the hyperparameter search
+# procedure. By default, the |RandomizedSearchCV| can provide those results as
+# a dictionary with somewhat complicated keys.
+#
+# We can ask the |Recipe| to create a nicer display of those results, using the
+# human-readable labels we provided earlier for the different hyperparameter
+# choices.
+
+# %%
+recipe.get_cv_results_table(randomized_search)
```

Comment:
I'm also curious to see whether people using hp tuning libraries like optuna or hyperopt could use the recipe easily, provided we know how to extract some sort of `cv_results_` from their tuners.

People could try something along the lines of:
```python
model = recipe.get_pipeline()
tuner.fit(model, recipe.get_X(), recipe.get_y())
recipe.plot_parallel(tuner)
```
Of course, that would require us to know the methods used by other libraries, but it could be worth it in a subsequent iteration. WDYT?